src/panel/category/CategoryPanel.jsx

```diff
@@ -149,29 +150,24 @@ export default class CategoryPanel extends React.Component {
       if (hasError) {
         panelContent = <CategoryPanelError zoomIn={zoomIn} />;
       } else {
-        panelContent = <PoiItemList
+        panelContent =
+        <PoiItemList
           pois={pois}
           selectPoi={this.selectPoi}
           highlightMarker={this.highlightPoiMarker}
+          dataSource={dataSource}
```

Comment:
It's a bit sad to have to change PoiItemList for this specific stuff. As a result you have to maintain it at 2 different levels in the tree. A best way would be to add the block here, in a `Fragment` with the `PoiItemList` element.